tools/build-toolchain.sh

```diff
@@ -1,34 +1,74 @@
 #! /bin/bash
-# N64 MIPS GCC toolchain build/install script for Unix distributions
+# N64 MIPS GCC toolchain build/install script (runs under UNIX systems).
 # (c) 2012-2021 Shaun Taylor and libDragon Contributors.
 # See the root folder for license information.
 
 
-
-# Before calling this script, make sure you have GMP, MPFR and TexInfo
-# packages installed in your system.  On a Debian-based system this is
-# achieved by typing the following commands:
+# Before calling this script, make sure you have all required
+# dependency packages installed in your system.  On a Debian-based systems
+# this is achieved by typing the following commands:
 #
-# sudo apt-get install libmpfr-dev
-# sudo apt-get install texinfo
-# sudo apt-get install libmpc-dev
+# sudo apt-get update && sudo apt-get upgrade
+# sudo apt-get install -yq wget bzip2 gcc g++ make file libmpfr-dev libmpc-dev zlib1g-dev texinfo git gcc-multilib
+
+# # Bash strict mode http://redsymbol.net/articles/unofficial-bash-strict-mode/
+# set -euo pipefail
+# IFS=$'\n\t'
 
-# Exit on error
+# Exit script on error
 set -e
 
-# Set N64_INST before calling the script to change the default installation directory path
-INSTALL_PATH="${N64_INST:-/usr/local}"
-# Set PATH for newlib to compile using GCC for MIPS N64 (pass 1)
-export PATH="$PATH:$INSTALL_PATH/bin"
+# Ensure you set 'N64_INST' before calling the script to change the default installation directory path
+  # by default it will presume 'usr/local/n64_toolchain'
+  INSTALL_PATH="${N64_INST:-/usr/local/n64_toolchain}"
+
+# Check for cross compile script flag
+# TODO: detect the flag `--host`
+# TODO: does not reach bash strict as $1 could be null!
+if [ "$1" == "x86_64-w64-mingw32" ]; then # Windows cross compile (host) flag is specified as a parameter.
+  # This (may) also require the following (extra) package dependencies:
+  # sudo apt-get install -yq mingw-w64 libgmp-dev bison libz-mingw-w64-dev autoconf
+
+  echo "Cross compiling for different host"
+  # Use the current-directory/binaries for the install path, as this is not for linux!
+  rm -rf binaries && mkdir binaries # always ensure the folder is clean (if rebuilding)
+  THIS_SCRIPT_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+  FOREIGN_INSTALL_PATH="$THIS_SCRIPT_PATH/binaries"
```

Comment:
I'm a little confused by this FOREIGN_INSTALL_PATH. Can you explain why we need this? The comment says "as this is not for linux", but I don't understand why the Canadian cross build should need this special directory.

Reply:
I have improved this in the script to make it more explicit. But, it really comes down to the fact that we might be building for 2 different "HOST"s in the same script. The install path for the "native" system should indeed be installed to a defined path, but if we are doing a foreign install, should we be adding extra things there... Although on further thought, it is not doing anything different to what I have done previously so might revisit once the current build completes.